main.cpp

```diff
@@ -54,51 +54,89 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* name);   //1
+
+    int value; //2
+    std::string name;//3
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* constructorName) :
+value(v),
+name(constructorName)
+{}
+
+struct CompareObjects                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
+        if ( a == nullptr || b == nullptr)
+        {
+            std::cout << "Your pointer is not valid!" << std::endl;
+            return nullptr;
+        }
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
         return nullptr;
     }
 };
 
+
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float uValue1 { 0 }, uValue2 { 0 };
+    float reduceDistanceMember(float* updatedValue)
     {
-        
-    }
+        if (updatedValue == nullptr)
+        {
+            std::cout << "Your pointer is not valid!" << std::endl;
+            return 0;
+        }
+        std::cout << std::endl;
+        std::cout << "U's uValue1 value: " << this->uValue1 << std::endl;
+        this->uValue1 = *updatedValue;
+        std::cout << "U's uValue1 updated value: " << this->uValue1 << std::endl;
+        std::cout << "U's uValue2 initial value: " << this->uValue2 << std::endl;
+        while( std::abs(this->uValue2 - this->uValue1) > 0.001f )
+        {
+            this->uValue2 += 0.0001f;
+        }
+        std::cout << "U's uValue2 updated value: " << this->uValue2 << std::endl;
+        return this->uValue2 * this->uValue1;
+    }      //12
+    
 };
 
-struct <#structname2#>
+struct staticStruct
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float reduceDistance(U* that, float updatedValue)        //10
```

Comment:
`updatedValue` is a pointer type.
the type was originally:
`<#type name#>*`
make it a pointer, and then revise everything else accordingly.
- [ ] revise